modules/programs/pubs.nix

```
@@ -0,0 +1,58 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.pubs;
+
+  configFile = pkgs.writeText ".pubsrc" "${cfg.config}";
+
+in {
+  options.programs.pubs = {
+    enable = mkEnableOption "pubs";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.pubs;
+      defaultText = literalExpression "pkgs.pubs";
+      description = "The package to use for the pubs script.";
+    };
+
+    config = mkOption {
```

I would suggest `extraConfig` instead to better correspond with other similar options within Home Manager:
```suggestion
    extraConfig = mkOption {
```